src/gui/folder.cpp

```diff
@@ -314,6 +314,62 @@ void Folder::etagRetreivedFromSyncEngine(const QString &etag)
     _lastEtag = etag;
 }
 
+void Folder::slotAccountCapabilitiesChanged()
+{
+    const Capabilities& capabilities = _accountState->account()->capabilities();
+    QString wsUrl = capabilities.getWebSocketUrl();
+    if (wsUrl.isEmpty()) {
+        return;
+    }
+
+    if (_webSocket) {
+        return;
+    }
+    //initialize wamp connection
+    _webSocket.reset(new QTcpSocket());
+    QUrl webSocketUrl(wsUrl);
```

Comment:
You also probably need to handle the disconnect case, and re try to connect.  Also handle error such as timeout and other kind.
So you will need to move this connection code somewhere else so you can call it when you want to re-connect.